### PR TITLE
[action][ensure_env_vars] efficiency improvement

### PR DESCRIPTION
**Pull request by crazymanish**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
- **Problem:** If multiple ENV variables are missing then the user has to run `ensure_env_vars` multiple times because it always EXIT after found the first missing ENV variable. 
- This is slightly inconvenience because Fastlane uses lots of ENV variables and good to report **all expected** missing ENV variables after one run itself to save time and to avoid Fastlane run `ensure_env_vars` multiple times 😇

### Description
- Made `ensure_env_vars` more efficient, now it will report all missing ENV variables after one run itself.
- Please see the screenshot for more details

### Testing Steps
- Update Gemfile and run `bundle install` with
```
gem "fastlane", :git => "git@github.com:crazymanish/fastlane.git", :branch => "ensure-env-vars-improvement"
```
- With the below test lane, run `bundle exec fastlane test`

```
lane :test do
    #ENV["GITHUB_API_TOKEN_MISSING_EXAMPLE"] = 'xxxxx-xxxxxxx-xxxxxxx'
    ENV["GYM_SCHEME"] = "MyApp"
    ENV["GYM_EXPORT_METHOD"] = 'app-store'
    #ENV["GYM_CLEAN"] = 'true'
    ENV["GYM_SILENT"] = 'true'
    #ENV["SLACK_API_TOKEN_MISSING_EXAMPLE"] = 'xxxxx-xxxxxxx-xxxxxxx'

    ensure_env_vars(
      env_vars: [
        'GITHUB_API_TOKEN_MISSING_EXAMPLE',
        'GYM_SCHEME',
        'GYM_EXPORT_METHOD',
        'GYM_CLEAN',
        'GYM_SILENT',
        'SLACK_API_TOKEN_MISSING_EXAMPLE'
      ]
    )
  end
```

### Screenshot
Screenshot 2021-04-08 at 20 26 22

### CODE REVIEW

File: fastlane/lib/fastlane/actions/ensure_env_vars.rb [hunk 1]
```diff
@@ -3,15 +3,12 @@ module Actions
...
         variables = params[:env_vars]
+        missing_variables = variables.select { |variable| ENV[variable].to_s.strip.empty? }
 
-        variables.each do |variable|
-          next unless ENV[variable].to_s.strip.empty?
-
-          UI.user_error!("Missing environment variable '#{variable}'")
-        end
+        is_one = missing_variables.length == 1
+        UI.user_error!("Missing environment variable#{is_one ? '' : 's'} '#{missing_variables.join('\', \'')}'") unless missing_variables.empty?
```
Comment (ainame) [inline fastlane/lib/fastlane/actions/ensure_env_vars.rb:9]:
```suggestion
        UI.user_error!("Missing environment variable(s) '#{missing_variables.join('\', \'')}'") unless missing_variables.empty?
```

I understand that this logic comes from the below line but isn't this enough here? 🙂 (and reusing the same local variable for a different purpose doesn't sound good in general).

Comment (crazymanish) [reply fastlane/lib/fastlane/actions/ensure_env_vars.rb]: This small logic `{is_one ? '' : 's'}` was breaking 2 existing unit tests, did not write this way in the first place. 
But, I agree with your suggestion, so applied it and updated the unit-tests too 🙇

**Review by ainame — APPROVED** [review]
🚀

**Review by joshdholtz — APPROVED** [review]
This is 🔥 Thanks for fixing this! Really appreciate all of your contributions ❤️

**Pull request merged**